src/Logging/test/ConsoleLoggerTest.cs

```diff
@@ -382,14 +382,50 @@ public void WriteCore_LogsCorrectTimestamp(ConsoleLoggerFormat format, LogLevel
                     Assert.Matches("^\\d{14}\\s$", sink.Writes[0].Message);
                     Assert.StartsWith(levelPrefix, sink.Writes[1].Message);
                 }
-                break;
+                    break;
                 case ConsoleLoggerFormat.Systemd:
                 {
                     Assert.Single(sink.Writes);
                     Assert.Matches("^<\\d>\\d{14}\\s[^\\s]", sink.Writes[0].Message);
                     Assert.StartsWith(levelPrefix, sink.Writes[0].Message);
                 }
-                break;
+                    break;
+                default:
+                    throw new ArgumentOutOfRangeException(nameof(format));
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(FormatsAndLevels))]
+        public void WriteCore_LogsCorrectTimestampInUtc(ConsoleLoggerFormat format, LogLevel level)
+        {
+            // Arrange
+            var t = SetUp(new ConsoleLoggerOptions { TimestampFormat = "yyyyMMddHHmmss ", Format = format, UseUtcTimezone = true});
+            var levelPrefix = t.GetLevelPrefix(level);
+            var logger = t.Logger;
+            var sink = t.Sink;
+            var ex = new Exception("Exception message" + Environment.NewLine + "with a second line");
+
+            // Act
+            logger.Log(level, 0, _state, ex, _defaultFormatter);
+
+            // Assert
```

Comment:
Hm, there is no assert that UTC is written, so this test is more or less a duplication of the already existing test for normal datetime. (just see you commented on that too :wink:).

> Should I just deserialize them and verify the timezone?

That's possible (but then you have to assert / take into account that current timezone != UTC).

Maybe it's better to (at least in my projects I would do) extract the creation of the `timestamp` in `ConsoleLogger` to a separate method that is testable?

Something like 
```c#
private static string GetTimestamp(string timestampFormat, bool useUtcTimezone)
    => GetTimestamp(timestampFormat, useUtcTimezone, DateTime.Now);

internal static string GetTimestamp(string timestampFormat, bool useUtcTimezone, DateTime now)
{
    if (timestampFormat == null)
        return null;

    if (useUtcTimezone)
        now = now.ToUniversalTime();

    return now.ToString(timestampFormat);
}
```
Then the second method is testable.

Reply:
That's one approach but who says that method is being called? We have no guarantee about that actually so not sure if that's going to cover everything.

Reply:
You're right. So your approach is a better end-to-end test.

Reply:
So what I'd propose then would be:
1. Change format to `yyyyMMddHHmmsszzz`
2. Get tracked timestamp via regex similar to https://github.com/aspnet/Extensions/pull/1896/files#diff-a27bf784d2879ac5f61a0d1a8573fc80R418
3. Parse to DateTime and check timezone

Thoughts?

I'm not a regex expert so help would be ideal on that 😅

Reply:
This should be there, except for explicitly setting the culture to a non-UTC compatible one for the non-UTC test.

I tried to use the folling but it didn't work:
```csharp
var cultureInfo = new CultureInfo("en-US");
CultureInfo.DefaultThreadCurrentCulture = cultureInfo;
CultureInfo.DefaultThreadCurrentUICulture = cultureInfo;
```